### DISCUSSION TITLE
gnuplot-lua-tikz.sty with homebrew

### DISCUSSION THREAD

**Discussion by psharp15** [Homebrew/discussions#4277] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: ba6d87eed9ce31f42fa2b33a4e6067d9cf17f13f
Last commit: 8 days ago
Core tap JSON: 28 Feb 10:05 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I've install mactex and gnu plot with home-brew, but when trying to use latex in my gnu plot to make pretty plots, it's missing this file (gnuplot-lua-tikz.sty). My advisor manually installed gnu plot, and here are his instructions on how to fix this issue. How can I do this but with home brew?

"To generate the style files, go to the work directory and copy in a '.lua' script to
gnuplot-tikz.lua
Then run
lua gnuplot-tikz.lua style
Then I copy the style files to:
/usr/local/texlive/texmf-local/tex/latex/gnuplot/
and run:
sudo texhash"

Like, where is the gnuplot-tikz.lua directory in Cellar/gnuplot?

**Update!**
When I run my gnu plot script to plot, I got the following message:

`
! LaTeX Error: File `gnuplot-lua-tikz.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name: `

So I put in the following 4 files to the directory where my plotting script it, and it worked!!!! 
[Archive.zip](https://github.com/Homebrew/discussions/files/10849103/Archive.zip)

Where should I put these files so I don't have to copy them into every directory I plot from? Maybe here: /usr/local/texlive/2022/texmf-dist/tex/latex/ ?

Thanks!!

**Comment by carlocab** [comment, marked as answer]
> Like, where is the gnuplot-tikz.lua directory in Cellar/gnuplot?

You can try
```sh
find -H "$(brew --prefix gnuplot)" -name 'gnuplot-tikz.lua'
```
to show you where it is. You should be able to find it at
```
"$(brew --prefix gnuplot)/share/gnuplot/5.4/lua/gnuplot-tikz.lua"
```

**Reply by psharp15** [reply]
That worked!! Thank you very much :)

I'm now needing to copy the style files into my latex directory, but it says I don't have permission to write to it. Not a home brew problem though. 

Thanks for your help!

**Comment by medoasiri** [comment]
https://github.com/Homebrew/homebrew-core/commit/669ceea8eacfe793d03bc93413a3171153ff1575
